web/src/buddy_data.ts

```diff
@@ -91,7 +91,7 @@ export let user_matches_narrow = (
     stream_id: number | undefined,
 ): boolean => {
     if (stream_id) {
-        return stream_data.is_user_subscribed(stream_id, user_id);
+        return stream_data.is_user_loaded_and_subscribed(stream_id, user_id);
```

Comment:
This is fine because we're only displaying loaded users.

Reply:
Hmm, can you write an explanation for why we know the user is loaded? I'm unsure if this function needs a docstring defining when it is safe to call.

Reply:
`user_matches_narrow` is only called from the buddy list code, and is essentially saying whether we're displaying the user as subscribed to this narrow or not. We decided we'd load all recently active users specifically since those are most visible in the buddy list (and for large organizations, the only ones visible, I believe?). 

So I guess sometimes we might show an offline subscribed user in the wrong section of the buddy list, until the fetch is complete. (So I was maybe wrong about only displaying loaded users.)

Does all that sound right to you? I can add a docstring saying to only use it from within buddy list code, or only use it if you want data about what the buddy list is displaying (which is not necessarily the true value of if any user is subscribed to that narrow).

Reply:
Thinking about this more, I think we should investigate what we're doing for the buddy list some more before turning this on. [#api design > trimming subscriber data #34244 @ 💬](https://chat.zulip.org/#narrow/channel/378-api-design/topic/trimming.20subscriber.20data.20.2334244/near/2226456)

I'll do the rename for `user_matches_narrow_using_loaded_data` now though.